core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -375,6 +375,7 @@ object Uri {
     * Literal syntax for URIs.  Invalid or non-literal arguments are rejected
     * at compile time.
     */
+  @deprecated("""use uri"" string interpolation instead""", "0.20")
```

Comment:
As discussed, this one is going to cause the most migration pain, but it's like what we'll be doing in the long run.